packages/odyssey-react/src/components/Tooltip/Portal.tsx

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { FC, useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+
+export const Portal: FC = ({ children }) => {
+  const [el, setEl] = useState<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    setEl(document.createElement("div"));
+  }, []);
+
+  useEffect(() => {
+    if (el) {
+      document.body.appendChild(el);
+      return () => {
+        document.body.removeChild(el);
+      };
+    }
+    return;
```

Comment:
curious: is this explicit return enforced through this [TS compiler option?](https://www.typescriptlang.org/tsconfig#noImplicitReturns) or just a style choice? From what I understand this effect callback can return undefined and this wouldn't be necessary

Reply:
yes it was a lint error i had to add this to make it go away. not sure if there's better way

Reply:
useEffect(() => {
    const portalElement = el;
    if (portalElement) {
      document.body.appendChild(portalElement);
    }
    
    return () => {
       if (portalElement) {
        document.body.removeChild(portalElement);
      }
    };
}, [el]);